src/ble/BtpEngine.cpp

```diff
@@ -310,7 +310,7 @@ CHIP_ERROR BtpEngine::HandleCharacteristicReceived(System::PacketBufferHandle &&
         // mRxFragnentSize may be smaller than the characteristic size.  Make sure
         // we're not truncating to a data length smaller than what we have already consumed.
         VerifyOrExit(reader.OctetsRead() <= mRxFragmentSize, err = BLE_ERROR_REASSEMBLER_INCORRECT_STATE);
-        data->SetDataLength(chip::min(data->DataLength(), mRxFragmentSize));
+        data->SetDataLength(chip::min(data->DataLength(), static_cast<uint32_t>(mRxFragmentSize)));
```

Comment:
Please, everywhere, use size_t rather than uint32_t, if this is going to be a breaking change anyway?

Reply:
@tcarmelveilleux, I actually have considered size_t. The issue is that there are a lot of places in both APIs and storage that need modification. Using size_t potentially requires even more instances of static_casts in the code. Furthermore, the fact that sizeof(size_t) is different in different architectures(8 bytes in 64 bit LInux, e.g.) makes handling it difficult when comparing with different other variables(ptrdiff, ssize_t, etc). I am also hitting a bunch of unit test failures. So, my theory was that since it is based off the 4 byte max field in the TCP payload, using a uint32_t should be fine. 
What is the compelling argument in favor of size_t in this particular case when we know that the max size is lower than 4 bytes?

Reply:
> What is the compelling argument in favor of size_t 

The fact that all other buffer size things, spans, etc, use it.

Realistically, you can _store_ uint32_t in packetbuffer, _return_ size_t in the API.  API consumers that for some reason don't want to use size_t can check-and-cast, but for the most part things using buffer sizes should in fact be using size_t...

Reply:
>The fact that all other buffer size things, spans, etc, use it.

I believe they use it because the size of the buffers can be relatively open-ended and size_t allows them to pass in as big a size as the memory would allow it. That is not necessarily true in this case where the cap is set by the TCP payload size of 4 bytes. 
TBH, I am all for using size_t but am hitting a few issues in the Tests and also having to insert more static_casts in the code. Going to take a look again with size_t and dig into the issues.

Reply:
Changed all API parameters and returns of SystemPacketBuffer to size_t. Used static_cast at higher layers wherever applicable.